board/safety/safety_toyota.h

```diff
@@ -64,6 +64,17 @@ static bool toyota_get_quality_flag_valid(const CANPacket_t *to_push) {
 static void toyota_rx_hook(const CANPacket_t *to_push) {
   const int TOYOTA_LTA_MAX_ANGLE = 1657;  // EPS only accepts up to 94.9461
 
+  if (GET_BUS(to_push) == 2U) {
+    int addr = GET_ADDR(to_push);
+  
+    if (addr == 0x412) {
+      bool lkas_hud_visible = (GET_BYTE(to_push, 0U) & 0xC0U) > 0U;
+      if (lkas_hud_visible) {
+        lkas_button_press = MADS_BUTTON_PRESSED;
+      }
+    }
+  }
+
```

Comment:
This currently does not reference 6th and 7th bit, need to right-shift 6 positions.
```suggestion
  if (GET_BUS(to_push) == 2U) {
    int addr = GET_ADDR(to_push);
  
    if (addr == 0x412) {
      int lkas_hud_visible = (GET_BYTE(to_push, 0U) & 0xC0U) >> 6U;
      if (lkas_hud_visible > 0) {
        lkas_button_press = MADS_BUTTON_PRESSED;
      }
    }
  }

```